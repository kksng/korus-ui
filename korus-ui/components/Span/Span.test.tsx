import React from 'react';
import toJson from 'enzyme-to-json';
import { mount } from 'enzyme';
import { Span } from './index';

describe('Span SNAPSHOTS', () => {
  it('should render', () => {
    const wrapper = mount(<Span>{'Hi, I\'m a text!'}</Span>);

    expect(wrapper.find('span')).toHaveLength(1);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});

describe('Span HANDLERS', () => {
  it('should trigger onClick', () => {
    const onClick = jest.fn();
    const wrapper = mount(<Span onClick={onClick}>example</Span>);

    wrapper.find('span').props().onClick?.({} as React.MouseEvent);

    expect(onClick).toHaveBeenCalled();
  });
});

describe('Span ATTRIBUTES', () => {
  it('should have className, change classes through props and className should not change prop-classes', () => {
    const wrapper = mount(<Span _box>default</Span>);

    expect(wrapper.find('span').hasClass('box')).toBeTruthy();


    wrapper.setProps({ _active: true, _box: false });

    expect(wrapper.find('span').hasClass('box')).toBeFalsy();

    expect(wrapper.find('span').hasClass('active')).toBeTruthy();


    wrapper.setProps({ className: 'testClass' });

    expect(wrapper.find('span').hasClass('box')).toBeFalsy();

    expect(wrapper.find('span').hasClass('active')).toBeTruthy();

    expect(wrapper.find('span').hasClass('testClass')).toBeTruthy();
  });
});
