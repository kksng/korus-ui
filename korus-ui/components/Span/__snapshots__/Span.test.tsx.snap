// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Span SNAPSHOTS should render 1`] = `
<Span>
  <span>
    Hi, I'm a text!
  </span>
</Span>
`;
